tools/clientSimulator/pfClientSimulator.py

```diff
@@ -0,0 +1,208 @@
+
+    parser.add_argument("-c", "--coverage", action='store_true',
+                        help="generate coverage file at end of script")
+
+    args = parser.parse_args()
+
+    # Logging Configuration
+    logger = logging.getLogger(__name__)
+
+    # Decide what to write in console depending on verbose argument
+    consoleLogger = logging.StreamHandler()
+    if args.verbose:
+        consoleLogger.setLevel(logging.DEBUG)
+    else:
+        consoleLogger.setLevel(logging.INFO)
+    logger.addHandler(consoleLogger)
+
+    # The given directory should have a conf.json file
+    if not os.path.isfile(os.path.join(args.test_directory, "conf.json")):
+        # This error will only be logged in the terminal
+        logger.error(
+            "Cannot find configuration file : conf.json in {} directory.".format(
+                args.test_directory))
+        exit(1)
+
+    configParser = ConfigParser(
+        os.path.join(
+            args.test_directory,
+            "conf.json"),
+        args.test_directory,
+        consoleLogger)
+
+    # Always write all log in the file
+    logging.basicConfig(level=logging.DEBUG,
+                        format='%(name)-12s %(levelname)-8s %(message)s',
+                        filename=configParser["LogFile"],
+                        filemode='w')
+
+    # Parsing criterion file and classes generation
+    logger.info("Criterion analysis")
+    classFactory = CriterionClassFactory(configParser["CriterionFile"])
+    criterionClasses = classFactory.generateCriterionClasses()
+
+    # Tests Handlers Generation
+    testFactory = TestVectorFactory(
+        criterionClasses,
+        consoleLogger)
+
+    testLauncher = TestLauncher(
+        criterionClasses,
+        configParser,
+        consoleLogger)
+
+    # Initialisation
+    testLauncher.init(criterionClasses, args.verbose)
+
+    # Launching
+    try:
+        if args.interactive:
+            # Launch Interactive Mode with default criterions values
+            UserInteractor(
+                testLauncher,
+                testFactory.generateTestVector()).launchInteractiveMode()
+        else:
+            while True:
+                scenarioOptions = {
+                    scenarioNumber:
+                        (scenarioFileName,
+                         DynamicCallHelper(
+                             launchScenario,
+                             logger,
+                             consoleLogger,
+                             configParser["ActionGathererFile"],
+                             os.path.join(
+                                 configParser["ScenariosDirectory"], scenarioFileName),
+                             testFactory,
+                             testLauncher
+                         ))
+                    for scenarioNumber, scenarioFileName in enumerate(
+                        [file for file in sorted(os.listdir(
+                            configParser["ScenariosDirectory"]))])
```

Comment:
Not in any of the versions I checked (2.7 and 3.4 IIRC; but feel free to double-check).